tests/test_trans_linearize.py

```diff
@@ -121,11 +166,11 @@ def test_others(self):
         rhs = intvar(0, 10, name="r")
 
         cons = cp.max([a,b,c,5]) <= rhs
-        cons = only_var_lhs(cons)[0]
+        cons = linearize_constraint(cons)[0]
         self.assertEqual("(max(a,b,c,5)) <= (r)", str(cons))
 
         cons = cp.AllDifferent([a, b, c])
-        cons = only_var_lhs(cons)[0]
+        cons = linearize_constraint(cons)[0]
         self.assertEqual("alldifferent(a,b,c)", str(cons))
 
 
```

Comment:
Add a case `p.implies(x-2y+3z != ~w)`? I'm mainly interested in checking the nested `!=`...